elasticapm/instrumentation/packages/asyncio/starlette.py

```diff
@@ -0,0 +1,46 @@
+#  BSD 3-Clause License
+#
+#  Copyright (c) 2019, Elasticsearch BV
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#
+#  * Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+#  * Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+#  * Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+#  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+#  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+#  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+#  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+#  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+#  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from elasticapm import get_client
+from elasticapm.instrumentation.packages.asyncio.base import AsyncAbstractInstrumentedModule
+
+
+class StarletteServerErrorMiddlewareInstrumentation(AsyncAbstractInstrumentedModule):
+    name = "starlette"
+
+    instrument_list = [("starlette.middleware.errors", "ServerErrorMiddleware.__call__")]
```

Comment:
```suggestion
    name = "starlette"
    
    instrument_list = [("starlette.middleware.errors", "ServerErrorMiddleware.__call__")]
    
    # We defer creating the transaction to the registered middleware.
    # However, since we are instrumenting the `__call__` of the outermost ServerErrorMiddleware
    # we need this flag so that it is always instrumented, otherwise no transaction is started until the nested
    # ElasticAPM middleware is called.
    creates_transaction = True
```

There are still a couple test errors I get when I run with this locally, but this seems to solve the bulk of the problem.

Reply:
Good catch! Makes perfect sense that it's not calling the wrapped version when there's no transaction. I should have spotted that. I'll add this patch and see what I can do with the rest of the tests.